### ISSUE TITLE
proposal: shortform inline comment syntax

### ISSUE THREAD

**Issue by aubergine10** [eslint/eslint#6435]
Original discussion: https://github.com/eslint/eslint/issues/6431#issuecomment-226475184

Specifically:

Is one aspect of the problem caused by the verbosity of eslint inline comments, making them a chore to use?

Instead of doing something like this:

```javascript
/*eslint no-magic-numbers: "off"*/
```

There could be a shorthand `+` and `-` syntax to turn a rule on or off:

```javascript
// eslint -no-magic-numbers

// eslint +no-magic-numbers
```

Where:

* `-<rulename>` means turn rule `"off"`
* `+<rulename>` means revert to previous state, otherwise state defined in config file, otherwise `"error"`

If we also allowed aliases of rules, it would make toggling them even more brief:

```javascript
// in .eslintrc

//...
   aliases: {
     "nmn": "no-magic-numbers"
   }
```

An alias, when used in an inline comment, would always be prefixed with `@` to avoid possibility of conflicts with other rules. Also, it would allow a minified syntax to be used:

```javascript
'turn on no-magic-numbers:'
// +@nmn

'turn off no-magic-numbers:'
// -@nmn

'long rule name can be used instead of alias if desired:'
// +@no-magic-numbers
// -@no-magic-numbers

'now we have short form to refer to rule, we could do stuff like this:'
// @no-magic-numbers "error"
// @no-magic-numbers "off"

'maybe allow comma chaining, eg:'
// @one-var "warn", @no-magic-numbers "error"
```

If all keyboards have an `§` key (I've been Mac user for way too long), maybe that would be better than `@`?

**Comment by aubergine10** [comment]
With ultra-short alias syntax, we could allow line-specific toggles of rules, for example:

```javascript
setTimeout( /*...*/ ) // -@nmn

// or longer form

setTimeout( /*...*/ ) // -@no-magic-numbers
```

This would disable/enable rule for specific line of code, rather than block. To toggle rules at block level, the comment must be the only thing on that line, eg:

```javascript
function foo() {
  // -@no-magic-numbers

  // entire function/block now has that rule disabled
}
```

**Comment by platinumazure** [comment]
The way we currently handle lint suppression is pretty messy. Personally I would feel better tackling these after the current lint suppression code is refactored out of `lib/eslint.js`, but there hasn't been a lot of appetite for that when I last floated it to the team.

Basically, I currently think that code is pretty fragile (not saying it has bugs, just that any change is pretty risky) and a refactor would be good.

I'm not on the core team, so don't take this as a "No". :smile:

**Comment by ilyavolodin** [comment]
I think this would be helpful, in my opinion. I, personally, don't really use comments to disable rules in my code. I figured that if I want to get the effect I'm looking for when turning on the rule, I can live with a bit of discomfort. But I can see how it would save a lot of time for somebody. I definitely don't think we can do `// -@nmn` but `// -@no-magic-numbers` seems fine to me. Maybe even something like `/// no-magic-number` since it's pretty clear that you want to turn it off for this line, minus is not necessary.
However, as @platinumazure noted, the code is pretty messy right now, and we would probably want to refactor it out of eslint.js file before doing something like that. I think lack of appetite wasn't due to not supporting the idea, but mostly due to the lack of time to do it.

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
